recipes/si/all/conanfile.py

```diff
@@ -36,38 +38,47 @@ def _compilers_minimum_version(self):
             "apple-clang": "10",
         }
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "17")
+            check_min_cppstd(self, self._min_cppstd)
 
-        minimum_version = self._compilers_minimum_version.get(
-            str(self.settings.compiler), False)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
         if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("'si' requires C++17, which your compiler ({} {}) does not support.".format(
-                    self.settings.compiler, self.settings.compiler.version))
+            if Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    "'si' requires C++17, which your compiler "
+                    f"({self.settings.compiler} {self.settings.compiler.version}) does not support.")
         else:
-            self.output.warn(
-                "'si' requires C++17. Your compiler is unknown. Assuming it supports C++17.")
-
-    def package_id(self):
-        self.info.header_only()
+            self.output.warning("'si' requires C++17. Your compiler is unknown. Assuming it supports C++17.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SI_BUILD_TESTING"] = False
+        tc.variables["SI_BUILD_DOC"] = False
+        tc.variables["SI_INSTALL_LIBRARY"] = True
```

Comment:
Is there a good reason to use variables instead of cache variables?

```suggestion
        tc.cache_variables["SI_BUILD_TESTING"] = False
        tc.cache_variables["SI_BUILD_DOC"] = False
        tc.cache_variables["SI_INSTALL_LIBRARY"] = True
```

Reply:
``variables`` have an advantage over ``cache_variables``: they work easier for local-development, for all users using presets and using conan_toolchain, with the same result. Otherwise it is easier for local-development to build something different if they don't pass explicitly all those ``-DSI_BUILD_DOC=False`` on the command line.

But given that ConanCenter recipes are almost never developed locally, not a big deal.